src/libraries/Native/Unix/System.Native/pal_random.c

```diff
@@ -108,3 +108,77 @@ void SystemNative_GetNonCryptographicallySecureRandomBytes(uint8_t* buffer, int3
     }
 #endif // HAVE_ARC4RANDOM_BUF
 }
+
+/*
+
+Generate cryptographically strong random bytes.
+
+Return 0 on success, -1 on failure.
+*/
+int32_t SystemNative_GetCryptographicallySecureRandomBytes(uint8_t* buffer, int32_t bufferLength)
```

Comment:
Call this from `SystemNative_GetNonCryptographicallySecureRandomBytes` to avoid the duplication?

Reply:
I wanted to avoid mixing the secure and non-secure case, to avoid the possibility of somehow returning non-secure random data.

Reply:
If the NonSecure function calls Secure function, there should not be risk like this.

We should at least make sure that we have just one urandom handle. Each function will get its own the way it is written that is unnecessary.

Reply:
Got it now.